tripy/examples/resent50/tripy-res50.ipynb

```diff
@@ -0,0 +1,599 @@
+    "    def __init__(self):\n",
+    "        super(ResNetEmbeddings, self).__init__()\n",
+    "        self.embedder = ResNetConvLayer(3, 64, kernel_dims=(7, 7), stride=(2, 2), padding=((3, 3), (3, 3)))\n",
+    "        self.pooler = MaxPool2d(kernel_dims=(3, 3), stride=(2, 2), padding=((1, 1), (1, 1)))\n",
+    "\n",
+    "    def __call__(self, x):\n",
+    "        x = self.embedder(x)\n",
+    "        x = self.pooler(x)\n",
+    "        return x"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Define a custom adaptive average pooling layer and the resenet backbone model itself"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class AdaptiveAvgPool2d(tp.Module):\n",
+    "    def __init__(self, output_size):\n",
+    "        super(AdaptiveAvgPool2d, self).__init__()\n",
+    "        self.output_size = output_size\n",
+    "\n",
+    "    def __call__(self, x):\n",
+    "        N, C, H_in, W_in = x.shape\n",
+    "        H_out, W_out = self.output_size\n",
+    "\n",
+    "        # Calculate stride and kernel size\n",
+    "        stride_h = H_in // H_out\n",
+    "        stride_w = W_in // W_out\n",
+    "\n",
+    "        kernel_size_h = H_in - (H_out - 1) * stride_h\n",
+    "        kernel_size_w = W_in - (W_out - 1) * stride_w\n",
+    "\n",
+    "        return tp.avgpool(x, kernel_dims=(int(kernel_size_h), int(kernel_size_w)), stride=(int(stride_h), int(stride_w)))\n",
+    "\n",
+    "class ResNetModel(tp.Module):\n",
+    "    def __init__(self):\n",
+    "        super(ResNetModel, self).__init__()\n",
+    "        self.embedder = ResNetEmbeddings()\n",
+    "        layers_config = [\n",
+    "            (3, 256, 64, (1, 1)),\n",
+    "            (4, 512, 128, (2, 2)),\n",
+    "            (6, 1024, 256, (2, 2)),\n",
+    "            (3, 2048, 512, (2, 2)),\n",
+    "        ]\n",
+    "        self.encoder = ResNetEncoder(layers_config)\n",
+    "        self.pooler = AdaptiveAvgPool2d(output_size=(1, 1))\n",
+    "\n",
+    "    def __call__(self, x):\n",
+    "        x = self.embedder(x)\n",
+    "        x = self.encoder(x)\n",
+    "        x = self.pooler(x)\n",
+    "        return x\n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Define resenet classifier head and a wrapper class `TPResNetClassifier` to combine backbone model and classifier."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class Flatten(tp.Module):\n",
+    "    def __init__(self, start_dim=1, end_dim=-1):\n",
+    "        super(Flatten, self).__init__()\n",
+    "        self.start_dim = start_dim\n",
+    "        self.end_dim = end_dim\n",
+    "\n",
+    "    def __call__(self, x):\n",
+    "        return tp.flatten(x, start_dim=self.start_dim, end_dim=self.end_dim)\n",
```

Comment:
Let's just use `tp.flatten` instead of creating a module?

Reply:
done